### PR TITLE
UniformRandomFill for unified storage

### PR DESCRIPTION
**Pull request by szalpal**
Signed-off-by: Michał Szołucha <mszolucha@nvidia.com>

#### Why we need this PR?
- It adds new feature: uniformly random filling unified storage tensors

#### What happend in this PR?
 - Changed specified type to template type and add static assert to make sure it's accessible

### CODE REVIEW

File: dali/kernels/test/tensor_test_utils.h [hunk 1]
```diff
@@ -219,19 +222,21 @@ if_iterable<Collection, void> UniformRandomFill(
...
 }
 
-template <typename DataType, int ndim, typename RandomGenerator>
+template <typename StorageBackend, typename DataType, int ndim, typename RandomGenerator>
 void UniformRandomFill(
-    const TensorView<StorageCPU, DataType, ndim> &tv,
+    const TensorView<StorageBackend, DataType, ndim> &tv,
     RandomGenerator &generator,
     same_as_t<DataType> lo, same_as_t<DataType> hi) {
+  static_assert(is_cpu_accessible<StorageBackend>::value, "Function available only for CPU-accessible TensorView backend");
```
Comment (mzient) [inline dali/kernels/test/tensor_test_utils.h]: This message is convoluted.
```suggestion
  static_assert(is_cpu_accessible<StorageBackend>::value, "This function requires a CPU-acessible tensor.");
```

File: dali/kernels/test/tensor_test_utils.h [hunk 2]
```diff
@@ -184,13 +185,15 @@ if_iterable<Collection, void> Fill(Collection &&collection, Generator &generator
     x = generator();
 }
 
-template <typename DataType, int ndim, typename Generator>
-void Fill(const TensorView<StorageCPU, DataType, ndim> &tv, Generator &generator) {
+template <typename StorageBackend, typename DataType, int ndim, typename Generator>
+void Fill(const TensorView<StorageBackend, DataType, ndim> &tv, Generator &generator) {
+  static_assert(is_cpu_accessible<StorageBackend>::value, "Function available only for CPU-accessible TensorView backend");
```
Comment (mzient) [inline dali/kernels/test/tensor_test_utils.h]:
```suggestion
  static_assert(is_cpu_accessible<StorageBackend>::value, "This function requires a CPU-acessible tensor.");
```

**Comment by szalpal** [comment]
!build

File: dali/kernels/test/tensor_test_utils.h [hunk 3]
```diff
@@ -65,28 +62,36 @@ void CheckEqual(
...
+
+template<int dim1, int dim2>
 void CheckEqual(const TensorListShape<dim1> &s1, const TensorListShape<dim2> &s2) {
   int max_errors = detail::DefaultMaxErrors;
   CheckEqual(s1, s2, max_errors);
 }
 
-template <typename T>
+
+template<typename T>
```
Comment (klecki) [inline dali/kernels/test/tensor_test_utils.h]: Why do you remove the space between `template` and `<`? It's counterproductive, we have .clang-format that adds the spaces automatically  and this just makes the diffs bigger.

Comment (szalpal) [reply dali/kernels/test/tensor_test_utils.h]: My bad. Missed it in my autoformat conf

**Comment by szalpal** [comment]
!build

File: dali/kernels/test/tensor_test_utils.h [hunk 4]
```diff
@@ -48,8 +45,8 @@ void CheckEqual(
     if (s1.tensor_shape_span(i) != s2.tensor_shape_span(i)) {
       if (++errors < max_errors) {
         EXPECT_EQ(s1.tensor_shape_span(i), s2.tensor_shape_span(i))
-            << "Samples #" << i
-            << " have different shapes " << s1[i] << " vs " << s2[i];
+                      << "Samples #" << i
```
Comment (mzient) [inline dali/kernels/test/tensor_test_utils.h]: Excessive indent.

File: dali/kernels/test/tensor_test_utils.h [hunk 5]
```diff
@@ -65,28 +62,36 @@ void CheckEqual(
...
   CheckEqual(s1, s2, max_errors);
 }
 
+
 template <typename T>
 T printable(T t) { return t; }
 
+
 inline int printable(char c) { return c; }
+
```
Comment (mzient) [inline dali/kernels/test/tensor_test_utils.h]: Why? Please remove these extra lines - you can keep one between the functions if you insist, but two empty lines between one-liners is too much.

File: dali/kernels/test/tensor_test_utils.h [hunk 6]
```diff
@@ -243,27 +269,32 @@ if_iterable<C, void> ConstantFill(C &&c, const element_t<C> &value = {}) {
...
 }
 
-template <typename DataType, int dim>
-void ConstantFill(
-    const TensorView<StorageCPU, DataType, dim> &tv,
-    same_as_t<DataType> value = {}) {
+
+template <typename StorageBackend, typename DataType, int dim>
+void
+ConstantFill(const TensorView<StorageBackend, DataType, dim> &tv, same_as_t<DataType> value = {}) {
```
Comment (mzient) [inline dali/kernels/test/tensor_test_utils.h]: Maybe it's just me, but I find the former formatting more readable.

File: dali/kernels/test/tensor_test_utils.h (hunk 5 above)
Comment (szalpal) [reply dali/kernels/test/tensor_test_utils.h]: Who says, that it's too much? :)

**Comment by szalpal** [comment]
!build

**Pull request merged**